### PR TITLE
Add release notes for 0.220

### PR DESCRIPTION
**Pull request by caithagoras**

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
+=============
+Release 0.219
+=============
+
+General Changes
+---------------
+* Fix an issue when executing spatial join on top of a right join. See :issue:`12672`.
+* Fix an issue where ``TRY`` cannot catch the failures when ``ROW`` and ``ARRAY`` with ``NULLs`` fails to compare.
+* Lazily build the hash table for map.
```
Comment (caithagoras) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: @yingsu00 Could you provide more context here. For example, does it improve performance in certain cases?

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+Hive Connector Changes
+----------------------
+* Add ``compression_codec`` session property to set the compression codec when writing files. Possible values are ``NONE``, ``SNAPPY``, and ``GZIP``.
+* Add experimental support for partial merge pushdown, which optimizes plan for queries joining over two tables with mismatched but compatible bucket counts.
+  This feature can be enabled by session property ``experimental.optimizer.partial-merge-pushdown-strategy``, and is aiming to supersede ``hive.optimize-mismatched-bucket-count``.
+  See :issue:`12611`.
+* Improve writing to bucketed table by cache Hive Metastore.
+
+
+MongoDB Connector
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: MongoDB Connector Changes

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
+=============
+Release 0.219
+=============
+
+General Changes
+---------------
+* Fix an issue when executing spatial join on top of a right join. See :issue:`12672`.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: Clarify what kind of issue (that it's a query failure, rather than a correctness or performance issue). e.g.
```
Fix an issue where spatial joins executed after right outer joins would fail in certain cases.
```

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+* Fix an issue where ``TRY`` cannot catch the failures when ``ROW`` and ``ARRAY`` with ``NULL`` elements fail to compare.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]:
```
* Fix an issue where ``TRY`` would not catch failures when a ``ROW`` or ``ARRAY`` with ``NULL`` elements failed to compare.
```

I'm actually not sure what "failed to compare" means.  What's it comparing with?

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+* Lazily build the hash table for map.
+
+Raptor Changes
+--------------
+* Integrate the optimized ORC writer into Raptor. Add configuration property ``storage.orc.optimized-writer-stage`` to enable and configure the optimized writer.
+  To switch to the optimized ORC writer is only backward compatible but not forward compatible.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: The optimized ORC writer is backward compatible with the Hadoop ORC writer, but the Hadoop ORC writer is not forward compatible with it.

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  Turning on the optimized ORC writer should work on an existing Raptor server. However, switching it back to Hadoop ORC writer can fail Raptor background job.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: s/switching it back to Hadoop/switching back to the Hadoop/

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+* Integrate ``ZSTD`` compression for file writer. Add configuration property ``storage.orc.compression-kind`` to switch between ``SNAPPY`` and ``ZSTD``.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]:
```
Add support for writing files with ``ZSTD`` compression.  Add configuration ...
```
I'm assuming that "Integerate zstd compression for file writer" means adding support for writing zstd files, but I'm not actually certain.

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  This is only applied when optimized ORC writer is turned on.
+
+
+Hive Connector Changes
+----------------------
+* Add ``compression_codec`` session property to set the compression codec when writing files. Possible values are ``NONE``, ``SNAPPY``, and ``GZIP``.
+* Add experimental support for partial merge pushdown, which optimizes plan for queries joining over two tables with mismatched but compatible bucket counts.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: s/optimizes plan/optimizes plans

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+  This feature can be enabled by session property ``experimental.optimizer.partial-merge-pushdown-strategy``, and is aiming to supersede ``hive.optimize-mismatched-bucket-count``.
+  See :issue:`12611`.
+* Improve writing to bucketed table by cache Hive Metastore.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: what's improved? is this a user-facing change?

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+  Tag sets are separated by an ampersand, and each tag set is specified as a comma-separated list of colon-separated key-value pairs.
+  For example, ``mongodb.read-preference-tags=dc:east,use:reporting&use:reporting``.
+
+JDBC Driver Changes
+-------------------
+* Fix incorrect precision and column display size in ResultSetMetaData for char and varchar columns.
+
+Verifier Changes
+----------------
+* Fix an issue where Verifier always mark floating point columns as mismatched if one of the control checksum and the test checksum is 0 while the other is close to 0.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: Fix an issue where Verifier would always mark floating point columns as mismatched if either the control checksum or test checksum was 0 while the other was close to 0.

File: presto-docs/src/main/sphinx/release/release-0.220.rst (hunk 9 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: @picrinite Could you add more details about this?

Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: Changed to
```
Improve query performance by caching partition information to
avoid duplicate Hive Metastore calls when writing bucketed tables.
```

File: presto-docs/src/main/sphinx/release/release-0.220.rst (hunk 4 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]:
```
Fix an issue where ``TRY`` cannot catch the failures when 2 ``ROWs`` or 2 ``ARRAYs``
with ``NULL`` elements fail to compare.
```

File: presto-docs/src/main/sphinx/release/release-0.220.rst (hunk 9 above)
Comment (tdcmeehan) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: I think this would be too complicated for most people to parse.  How about

```
Improve performance of writes to bucketed Hive tables.
```

Comment (caithagoras) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: What about
```
Improve query performance when writing bucketed tables.
```

@rschlussel How do we balance succinctness vs. context/details?

Comment (rschlussel) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: I like @tdcmeehan's suggestion or your second suggestion. Our release notes focus on the user, so the implementation details don't necessarily matter, but we should always include context about what kinds of queries are affected and how.  
Here's what context that I feel is important to provide for some common scenarios:
* bug fixes -- what kinds of queries encountered the issue, what would the user have seen before (failures with a particular error message, wrong results, slow performance), and if it's not obvious, then say what the behavior is now.
* new features -- basic description of the feature and how to turn it on and link to docs about it
* performance improvement- what kinds of queries will see improved performance and possibly what will be improved about it (e.g. is it a walltime improvement or memory improvement, etc.).

File: presto-docs/src/main/sphinx/release/release-0.220.rst (hunk 1 above)
Comment (tdcmeehan) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: Something like improve performance in scan which contains maps

File: presto-docs/src/main/sphinx/release/release-0.220.rst [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+
+General Changes
+---------------
+* Fix an issue where spatial joins executed after right outer joins would fail in certain cases. See :issue:`12672`.
+* Fix an issue where ``TRY`` cannot catch the failures in ``ROW`` comparisons with ``NULL`` fields and ``ARRAY`` comparisons with ``NULL`` elements.
+* Lazily build the hash table for map.
+
+Raptor Changes
+--------------
+* Integrate the optimized ORC writer into Raptor. Add configuration property ``storage.orc.optimized-writer-stage`` to enable and configure the optimized writer.
```
Comment (tdcmeehan) [inline presto-docs/src/main/sphinx/release/release-0.220.rst]: A user will not know what "the optimized ORC writer" is.  Something like

`Improve performance of ORC write performance in Raptor.  Use configuration property `storage.orc.optimized-writer-stage`` to enable his performance enhancement.`

File: presto-docs/src/main/sphinx/release/release-0.220.rst (hunk 1 above)
Comment (yingsu00) [reply presto-docs/src/main/sphinx/release/release-0.220.rst]: @tdcmeehan thanks for the explanation! Presto builds hashtable for MapBlocks eagerly when constructing the MapBlock even it's not needed in the query. Building a hashtable could take up to 30-40% CPU of the scan cost on a map column. This commit defers the hashtable build to the time it's needed in SeekKey(). So I would rephrase the release note as:
```
Lazily build the hash tables for MapBlock until it's probed, or skip building the hash tables for queries that don't need to probe them.
```
